packages/flutter_goldens_client/lib/skia_client.dart

```diff
@@ -0,0 +1,222 @@
+  /// will only be called once for each instance of
+  /// [FlutterSkiaGoldFileComparator].
+  ///
+  /// The [workDirectory] parameter specifies the current directory that golden
+  /// tests are executing in, relative to the library of the given test. It is
+  /// informed by the basedir of the [FlutterSkiaGoldFileComparator].
+  Future<void> auth(Directory workDirectory) async {
+    assert(workDirectory != null);
+    _workDirectory = workDirectory;
+    if (_clientIsAuthorized())
+      return;
+
+    if (_serviceAccount.isEmpty) {
+      final StringBuffer buf = StringBuffer()..writeln('Gold service account is unavailable.');
+      throw NonZeroExitCode(1, buf.toString());
+    }
+
+    final File authorization = _workDirectory.childFile('serviceAccount.json');
+    await authorization.writeAsString(_serviceAccount);
+
+    final List<String> authArguments = <String>[
+      'auth',
+      '--service-account', authorization.path,
+      '--work-dir', _workDirectory.childDirectory('temp').path,
+    ];
+
+    // final io.ProcessResult authResults =
+    await io.Process.run(
+      _goldctl,
+      authArguments,
+    );
+    // TODO(Piinks): Re-enable after Gold flakes are resolved, https://github.com/flutter/flutter/pull/36103
+    // if (authResults.exitCode != 0) {
+    //   final StringBuffer buf = StringBuffer()
+    //     ..writeln('Flutter + Skia Gold auth failed.')
+    //     ..writeln('stdout: ${authResults.stdout}')
+    //     ..writeln('stderr: ${authResults.stderr}');
+    //   throw NonZeroExitCode(authResults.exitCode, buf.toString());
+    // }
+  }
+
+  /// Executes the `imgtest init` command in the goldctl tool.
+  ///
+  /// The `imgtest` command collects and uploads test results to the Skia Gold
+  /// backend, the `init` argument initializes the current test.
+  Future<void> imgtestInit() async {
+    final File keys = _workDirectory.childFile('keys.json');
+    final File failures = _workDirectory.childFile('failures.json');
+
+    await keys.writeAsString(_getKeysJSON());
+    await failures.create();
+    final String commitHash = await _getCurrentCommit();
+
+    final List<String> imgtestInitArguments = <String>[
+      'imgtest', 'init',
+      '--instance', 'flutter',
+      '--work-dir', _workDirectory.childDirectory('temp').path,
+      '--commit', commitHash,
+      '--keys-file', keys.path,
+      '--failure-file', failures.path,
+      '--passfail',
+    ];
+
+    if (imgtestInitArguments.contains(null)) {
+      final StringBuffer buf = StringBuffer();
+      buf.writeln('Null argument for Skia Gold imgtest init:');
+      imgtestInitArguments.forEach(buf.writeln);
+      throw NonZeroExitCode(1, buf.toString());
+    }
+
+    // final io.ProcessResult imgtestInitResult =
+    await io.Process.run(
+      _goldctl,
+      imgtestInitArguments,
+    );
+
+    // TODO(Piinks): Re-enable after Gold flakes are resolved, https://github.com/flutter/flutter/pull/36103
```

Comment:
Flaky behavior from the skia gold service needs to be resolved before these exceptions can be enabled.